packages/api/src/graphql/collection-index/query/exist-index.ts

```diff
@@ -0,0 +1,56 @@
+import { gql } from "@apollo/client";
+import { NetworkResult, handleRequest } from "../../../utils/network";
+import client from "../../client";
+
+const EXIST_INDEX = gql`
+  query IndexExist(
+    $databaseName: String!
+    $collectionName: String!
+    $indexName: String!
+  ) {
+    indexExist(
+      databaseName: $databaseName
+      collectionName: $collectionName
+      indexName: $indexName
+    )
+  }
+`;
+
+interface ExistIndexResponse {
+  existed: boolean;
+}
+
+export const existIndex = async (
+  databaseName: string,
+  collectionName: string,
+  indexName: string,
+  token: string
+): Promise<NetworkResult<{ exist: boolean }>> => {
+  return handleRequest(async () => {
+    const { data } = await client.query<{ indexExist: ExistIndexResponse }>({
```

Comment:
We should have two request methods:

1. Unauthorized request
2. Authorized request

Reply:
I do not think it is necessary for this implementation, it is inconvenient to abstract it